src/Gui/ViewProviderDocumentObject.cpp

```diff
@@ -401,9 +401,9 @@ Gui::MDIView* ViewProviderDocumentObject::getActiveView() const
 {
     if(!pcObject)
         throw Base::RuntimeError("View provider detached");
-    App::Document* pAppDoc = pcObject->getDocument();

```

Comment:
A view provider is not supposed to return the active view of the application but the view of the document this view provider is part of.

With this change it will happen that the view of a different document could be returned.

Reply:
Alright i see. I've changed it to a simple check if the object is attached to a document because otherwise i get a segfault when calling ViewProviderDocumentObject::getActiveView.
This makes me curious, is it generally supported to have unattached DocumentObjects in the first place?